src/sentry/static/sentry/app/components/events/eventEntries.tsx

```diff
@@ -45,39 +65,174 @@ type Props = {
+        exceptionValue.stacktrace?.frames?.find(frame =>
+          this.isDataMinified(frame.module)
+        )
+      );
+    }
+
+    const threadExceptionValues = getThreadException(event, bestThread)?.values;
+
+    return !!(threadExceptionValues
+      ? threadExceptionValues.find(threadExceptionValue =>
+          threadExceptionValue.stacktrace?.frames?.find(frame =>
+            this.isDataMinified(frame.module)
+          )
+        )
+      : bestThread?.stacktrace?.frames?.find(frame => this.isDataMinified(frame.module)));
+  }
+
+  async checkProGuardError() {
     const {event} = this.props;
 
-    if (!event || !event.errors || !(event.errors.length > 0)) {
+    if (!event) {
       return;
     }
-    const errors = event.errors;
-    const errorTypes = errors.map(errorEntries => errorEntries.type);
-    const errorMessages = errors.map(errorEntries => errorEntries.message);
 
-    this.recordIssueError(errorTypes, errorMessages);
-  }
+    const hasEventErrorsProGuardMissingMapping = event.errors?.find(
+      error => error.type === 'proguard_missing_mapping'
+    );
 
-  shouldComponentUpdate(nextProps: Props) {
-    const {event, showExampleCommit} = this.props;
+    if (hasEventErrorsProGuardMissingMapping) {
+      this.setState({isLoading: false});
+      return;
+    }
 
-    return (
-      (event && nextProps.event && event.id !== nextProps.event.id) ||
-      showExampleCommit !== nextProps.showExampleCommit
+    const proGuardErrors: ProGuardErrors = [];
+
+    const debugImages = event.entries?.find(e => e.type === EntryType.DEBUGMETA)?.data
+      .images as undefined | Array<Image>;
+
+    // When debugImages contains a 'proguard' entry, it must always be only one entry
+    const proGuardImage = debugImages?.find(
+      debugImage => debugImage?.type === 'proguard' && !!debugImage?.uuid
     );
+
+    // If an entry is of type 'proguard' and has 'uuid',
+    // it means that the Sentry Gradle plugin has been executed,
+    // otherwise the proguard id wouldn't be in the event.
+    // But maybe it failed to upload the mappings file
+    if (proGuardImage?.uuid) {
+      const debugFile = await this.fetchDebugFile(proGuardImage.uuid);
+
+      if (!debugFile.length) {
+        proGuardErrors.push({
+          type: 'proguard_missing_mapping',
+          message: projectProcessingIssuesMessages.proguard_missing_mapping,
+          data: {mapping_uuid: proGuardImage.uuid},
+        });
+      }
+
+      this.setState({proGuardErrors, isLoading: false});
+      return;
+    }
+
+    const threads: Array<Thread> =
+      event.entries?.find(e => e.type === EntryType.THREADS)?.data?.values ?? [];
+
+    const bestThread = findBestThread(threads);
+    const hasThreadOrExceptionMinifiedData = this.hasThreadOrExceptionMinifiedFrameData(
+      event,
+      bestThread
+    );
+
+    if (hasThreadOrExceptionMinifiedData) {
+      proGuardErrors.push({
+        type: 'proguard_incorrectly_configured_plugin',
```

Comment:
It would be good to point out this is a best effort thing that will have some false positives and require fine tuning.

```suggestion
        type: 'proguard_potentially_misconfigured_plugin',
```